drivers/include/bq2429x.h

```diff
@@ -0,0 +1,500 @@
+ * with the @ref bq2429x_init_int functions.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the BQ2429x Single-Cell USB Charger
+ *
+ * @author      Jean Pierre Dudey <jeandudey@hotmail.com>
+ */
+
+#ifndef BQ2429X_H
+#define BQ2429X_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "kernel_defines.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Driver return values
+ */
+enum {
+    BQ2429X_OK              = 0,        /**< Everything is ok */
+    BQ2429X_ERR_I2C         = -1,       /**< I2C error */
+    BQ2429X_ERR_GPIO        = -2,       /**< GPIO initialization error */
+};
+
+/**
+ * @brief   Used VBUS type.
+ */
+typedef enum {
+    BQ2429X_VBUS_NONE   = 0,            /**< No input source detected */
+    BQ2429X_VBUS_USB_CHARGER,           /**< USB charger input */
+    BQ2429X_VBUS_AP_CHARGER,            /**< Adapter port charge input */
+    BQ2429X_VBUS_OTG,                   /**< VBUS is used for OTG */
+} bq2429x_vbus_stat_t;
+
+/**
+ * @brief   Battery charging status
+ */
+typedef enum {
+    BQ2429X_CHARGE_NONE = 0,            /**< The device is not charging a
+                                             battery */
+    BQ2429X_CHARGE_PRE,                 /**< Pre-charge state */
+    BQ2429X_CHARGE_FAST,                /**< Fast charging */
+    BQ2429X_CHARGE_DONE,                /**< Charging done */
+} bq2429x_chrg_stat_t;
+
+/**
+ * @brief   Device status
+ */
+typedef struct {
+    bq2429x_vbus_stat_t vbus;   /**< VBUS status */
+    bq2429x_chrg_stat_t chrg;   /**< Charge status */
+    /**
+     * @brief   DPM status
+     * @details false = Not in DPM.
+     *          true = VINDPM or IINDMP.
+     */
+    bool dpm;
+    /**
+     * @brief   Power good status
+     * @details false = Not Power Good
+     *          true  = Power Good
+     */
+    bool pg;
+    /**
+     * @brief   Thermal regulation
+     * @details false = Normal
+     *          true  = In thermal regulation
+     */
+    bool therm;
+    /**
+     * @brief   VSYSMIN regulation
+     * @details false = Not in VSYSMIN regulation (BAT voltage > VSYSMIN)
+     *          true  = In VSYSMIN regulation (BAT voltage < VSYSMIN)
+     */
+    bool vsys;
+} bq2429x_status_t;
+
+/**
+ * @brief   Charge fault values
+ */
+typedef enum {
+    BQ2429x_CHRG_FAULT_NORMAL = 0,              /**< No fault, normal */
+    BQ2429x_CHRG_FAULT_INPUT,                   /**< Input fault (OVP or bad
+                                                     source) */
+    BQ2429x_CHRG_FAULT_THERMAL_SHUTDOWN,        /**< Thermal shutdown */
+    BQ2429x_CHRG_FAULT_CHARGE_TIMER_EXPIRATION, /**< Charge timer expiration */
+} bq2429x_chrg_fault_t;
+
+/**
+ * @brief   Device faults
+ */
+typedef struct {
+    /**
+     * @brief   Watchdog fault.
+     * @details false = Normal.
+     *          true  = Watchdog timer expiration.
```

Comment:
Can the Watchdog timer be set by the user?

Reply:
Yes (just not exposed right now), the watchdog is for the I2C interface, by default has a 40s timeout. The watchdog timer starts when an I2C command is written, it will keep running until the timeout is reached and the device will transition from "controlled by host" mode to "default mode" again. When the timeout is reached that field is set.

Changing the watchdog timer timeout is only useful if constantly sending I2C commands to the device that will take more than the current timeout.